csharp/src/Google.Protobuf/CodedInputStream.cs

```diff
@@ -851,14 +1277,28 @@ internal static uint ReadRawVarint32(Stream input)
         /// <summary>
         /// Reads a raw varint from the stream.
         /// </summary>
+        [SecuritySafeCritical]
         internal ulong ReadRawVarint64()
         {
+            var immediateBudder = ImmediateBuffer;
+            return ReadRawVarint64(ref immediateBudder);
+        }
+
+        /// <summary>
+        /// This supports the Protocol Buffers infrastructure and is not meant to be used directly from your code.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        [SecurityCritical]
+        internal ulong ReadRawVarint64(ref ReadOnlySpan<byte> immediateBuffer)
+        {
+            //TODO: Implement fast path

```

Comment:
please explain in more detail if you want to leave this TODO.

Reply:
Hmmm ok let’s remove it for now. But it seemed quite tempting to use the immediateBuffer directly.